doc/tfsformat.rst

```diff
@@ -0,0 +1,142 @@
+
+.. code-block::
+
+    *             INTEGERS              STRINGS               FLOATS
+    $                   %d                   %s                  %le
+                         1            SOMETHING              4.51395
+                     21345         "WITH SPACE"             123.4825
+
+Types and Identifiers
+---------------------
+
+Type identifiers are specific strings that indicate as which type to interpret the given data as, in the individual entries of the `Headers`_ as well as for the columns of the `Table`_.
+The following type identifiers and their corresponding Python types are accepted and used by `tfs-pandas`:
+
+================  ======================  =============== =========================
+Type Indentifier  Associated Python Type          Example               Accepted by
+================  ======================  =============== =========================
+%s                                string        `%s name`  ``MAD-X`` and ``MAD-NG``
+%bpm_s                            string    `%bpm_s name`  ``MAD-X`` and ``MAD-NG``
+%d                        64-bit integer         `%d 174`  ``MAD-X`` and ``MAD-NG``
+%hd                       64-bit integer        `%hd 174`  ``MAD-X`` and ``MAD-NG``
+%f                          64-bit float       `%f 0.946`  ``MAD-X`` and ``MAD-NG``
+%le                         64-bit float      `%hf 0.946`  ``MAD-X`` and ``MAD-NG``
+%b                               boolean        `%b true`           Only ``MAD-NG``
+%lz                      128-bit complex   `%lz 1.4+2.6i`           Only ``MAD-NG``
+%n                              NoneType         `%n nil`           Only ``MAD-NG``
+================  ======================  =============== =========================
+
+It is also possible to include the length of a string into its type identifier, for instance `%10s` for a string of 10 characters.
+This was meant for older codes to know how much memory to allocate.
+While ``MAD-X`` and ``MAD-NG`` inlcude this information in their output **TFS** files, they do not require it when reading.
+For these reasons, by default `tfs-pandas` ignores the information when parsing files, and does not include it when writing them.
+
+.. admonition:: MAD-NG Specific Types
+
+    As can be seen in the table above, some types are specific to the ``MAD-NG`` code, such as boolean and complex types, and would not be accepted by ``MAD-X``.
+    Please see the :doc:`compatibility section <compatibility>` for more information.
+
+.. _tfs-pandas caveats:
+
+TFS-Pandas Caveats
+------------------
+
+The following caveats apply to the `tfs-pandas` package:
+
+- Column names should be strings.
+- No spaces should be present in column names.
+- The table should not contain duplicate indices.
+- The table should not contain duplicate columns.
+- If spaces are present in strings, they should be enclosed in either single or double quotes.
+- The table data should not contain nested structures (lists, tuples, etc.).
```

Comment:
What happens when a user has created a tfs dataframe through pymadng and then tries to use tfs.write(...). Will you error, warn or ignore?

In theory, you shouldn't do this. I.e. You can just use pymadng to read and write all these tfs dataframes, but what if someone tries...

Reply:
I'll be honest, I have no idea exactly. Mostly likely is an error being raised.

I agree that in theory one should **not** do this. I don't particularly intend to cater to that use case. Eager for @JoschD's opinion